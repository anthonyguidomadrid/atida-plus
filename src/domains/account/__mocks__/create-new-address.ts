export const createNewAddress = {
  redirect: 'some-path',
  salutation: 'string',
  firstName: 'string',
  lastName: 'string',
  address1: 'string',
  address3: 'string',
  zipCode: 'string',
  city: 'string',
  country: 'string',
  iso2Code: 'string',
  phone: 'string',
  isDefaultShipping: true,
  isDefaultBilling: true
}

export const createNewAddressPayload = {
  data: {
    type: 'customers',
    attributes: {
      salutation: 'string',
      firstName: 'string',
      lastName: 'string',
      address1: 'string',
      address3: 'string',
      zipCode: 'string',
      city: 'string',
      country: 'string',
      iso2Code: 'string',
      phone: 'string',
      isDefaultShipping: true,
      isDefaultBilling: true
    }
  }
} as const

export const UpdatedJWTAndRefreshTokens = {
  refreshToken: 'newRefreshToken',
  JWT: 'newJWT',
  expiresIn: '12345'
}

export const sprykerUpdatedJWTAndRefreshTokens = {
  data: {
    type: '',
    id: '',
    attributes: {
      tokenType: 'Bearer',
      expiresIn: '12345',
      accessToken: 'newJWT',
      refreshToken: 'newRefreshToken',
      idCompanyUser: ''
    },
    links: {
      self: ''
    }
  },
  links: {
    self: ''
  }
}
