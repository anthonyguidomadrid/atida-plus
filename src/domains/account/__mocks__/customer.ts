export const createCustomer = {
  salutation: 'Mr',
  firstName: 'Test',
  lastName: 'McTest',
  email: 'some@email.com',
  password: 'somePassword1!',
  confirmPassword: 'somePassword1!',
  acceptedTerms: true,
  emailNotification: true,
  phone: '654654654',
  company: null,
  surcharge: false,
  addresses: {
    addresses: [
      {
        salutation: 'Mr',
        firstName: 'Test',
        lastName: 'McTest',
        address1: 'Brugstraat 123',
        zipCode: '1011 AD',
        city: 'Amsterdam',
        iso2Code: 'PT',
        phone: '088273791298432',
        isDefaultShipping: true,
        isDefaultBilling: true
      }
    ]
  }
}

export const sprykerCustomer = {
  data: {
    type: 'customers',
    id: 'some-id',
    attributes: {
      salutation: 'Mr',
      firstName: 'Test',
      lastName: 'McTest',
      email: 'some@email.com',
      password: 'somePassword1!',
      confirmPassword: 'somePassword1!',
      acceptedTerms: true,
      emailNotification: true,
      company: null,
      surcharge: false
    }
  }
} as const

export const sprykerCustomerWithAddresses = {
  data: {
    ...sprykerCustomer,
    attributes: {
      ...sprykerCustomer.data.attributes
    },
    relationships: {
      addresses: {
        data: [
          { type: 'addresses', id: 'address-id-1' },
          { type: 'addresses', id: 'address-id-2' }
        ]
      }
    }
  },
  included: [
    {
      type: 'addresses',
      id: 'address-id-1',
      attributes: {
        salutation: 'some-salutation',
        firstName: 'some-name',
        lastName: 'some-last-name',
        address1: 'some-address',
        address2: 'some-address2',
        address3: 'some-address3',
        zipCode: 'some-zip-code',
        city: 'some-city',
        country: 'some-country',
        iso2Code: 'some-iso-code',
        company: 'some-company',
        phone: 'some-phone',
        isDefaultShipping: true,
        isDefaultBilling: true,
        isTaxExempt: false
      }
    },
    {
      type: 'addresses',
      id: 'address-id-2',
      attributes: {
        salutation: 'some-other-salutation',
        firstName: 'some-other-name',
        lastName: 'some-other-last-name',
        address1: 'some-other-address',
        address2: 'some-other-address2',
        address3: 'some-other-address3',
        zipCode: 'some-other-zip-code',
        city: 'some-other-city',
        country: 'some-other-country',
        iso2Code: 'some-other-iso-code',
        company: 'some-other-company',
        phone: 'some-other-phone',
        isDefaultShipping: false,
        isDefaultBilling: false,
        isTaxExempt: false
      }
    }
  ]
}
