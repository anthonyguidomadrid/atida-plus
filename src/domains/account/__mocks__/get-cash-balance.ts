export const balance = {
  total: 20
}
