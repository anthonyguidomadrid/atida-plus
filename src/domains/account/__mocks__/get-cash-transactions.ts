import { AtidaCashTransactionHistoryLedgerResponse } from '../types'

export const mockedTransactionHistory = {
  data: [
    {
      transaction_type: 'debit',
      amount: 200,
      transaction_date: '2022-10-04T14:20:22.029764445Z'
    },
    {
      transaction_type: 'credit',
      amount: 280,
      expiry_date: '2023-10-04T14:20:22.029764445Z',
      transaction_date: '2022-10-04T14:20:22.029764445Z'
    },
    {
      transaction_type: 'credit',
      amount: 280,
      expiry_date: '2023-07-23T14:20:22.029764445Z',
      transaction_date: '2022-07-23T14:20:22.029764445Z'
    },
    {
      transaction_type: 'debit',
      amount: 200,
      transaction_date: '2022-06-23T14:20:22.029764445Z'
    },
    {
      transaction_type: 'credit',
      amount: 280,
      expiry_date: '2023-06-23T14:20:22.029764445Z',
      transaction_date: '2022-06-23T14:20:22.029764445Z'
    },
    {
      transaction_type: 'credit',
      amount: 500,
      expiry_date: '2023-05-23T14:20:22.029764445Z',
      transaction_date: '2022-05-23T14:20:22.029764445Z'
    },
    {
      transaction_type: 'debit',
      amount: 280,
      expiry_date: '2023-04-23T14:20:22.029764445Z',
      transaction_date: '2022-04-23T14:20:22.029764445Z'
    },
    {
      transaction_type: 'credit',
      amount: 200,
      expiry_date: '2023-04-23T14:20:22.029764445Z',
      transaction_date: '2022-04-23T14:20:22.029764445Z'
    }
  ],
  metadata: {
    count: 8,
    total: 27
  }
} as AtidaCashTransactionHistoryLedgerResponse

export const mockedTransactionHistoryBayDateArray = {
  transactions: [
    {
      date: '2022-09-30',
      transactions: [
        {
          transaction_date: '2022-09-30T13:36:27.297Z',
          transaction_type: 'debit',
          amount: 200
        },
        {
          amount: 280,
          expiry_date: '2023-09-30T13:36:27.297Z',
          transaction_date: '2022-09-30T13:36:27.297Z',
          transaction_type: 'credit'
        },
        {
          transaction_type: 'debit',
          amount: 200,
          transaction_date: '2022-09-30T13:36:27.297Z'
        },
        {
          transaction_date: '2022-09-30T13:36:27.297Z',
          transaction_type: 'credit',
          amount: 180,
          expiry_date: '2023-09-30T13:36:27.297Z'
        },
        {
          amount: 100,
          transaction_date: '2022-09-30T13:36:27.297Z',
          transaction_type: 'debit'
        },
        {
          transaction_date: '2022-09-30T13:36:27.297Z',
          transaction_type: 'credit',
          amount: 670,
          expiry_date: '2023-09-30T13:36:27.297Z'
        }
      ]
    },
    {
      date: '2022-07-23',
      transactions: [
        {
          transaction_date: '2022-07-23T14:20:22.029764445Z',
          transaction_type: 'debit',
          amount: 200
        },
        {
          expiry_date: '2023-07-23T14:20:22.029764445Z',
          transaction_date: '2022-07-23T14:20:22.029764445Z',
          transaction_type: 'credit',
          amount: 280
        },
        {
          transaction_date: '2022-07-23T14:20:22.029764445Z',
          transaction_type: 'debit',
          amount: 200
        },
        {
          amount: 290,
          expiry_date: '2023-07-23T14:20:22.029764445Z',
          transaction_date: '2022-07-23T14:20:22.029764445Z',
          transaction_type: 'credit'
        }
      ]
    },
    {
      date: '2022-06-23',
      transactions: [
        {
          transaction_type: 'debit',
          amount: 200,
          transaction_date: '2022-06-23T14:20:22.029764445Z'
        },
        {
          expiry_date: '2023-06-23T14:20:22.029764445Z',
          transaction_date: '2022-06-23T14:20:22.029764445Z',
          transaction_type: 'credit',
          amount: 280
        }
      ]
    },
    {
      date: '2022-05-23',
      transactions: [
        {
          transaction_type: 'credit',
          amount: 500,
          expiry_date: '2023-05-23T14:20:22.029764445Z',
          transaction_date: '2022-05-23T14:20:22.029764445Z'
        }
      ]
    },
    {
      date: '2022-04-23',
      transactions: [
        {
          transaction_date: '2022-04-23T14:20:22.029764445Z',
          transaction_type: 'debit',
          amount: 280,
          expiry_date: '2023-04-23T14:20:22.029764445Z'
        },
        {
          transaction_type: 'credit',
          amount: 200,
          expiry_date: '2023-04-23T14:20:22.029764445Z',
          transaction_date: '2022-04-23T14:20:22.029764445Z'
        }
      ]
    }
  ],
  totalPages: 4
}
