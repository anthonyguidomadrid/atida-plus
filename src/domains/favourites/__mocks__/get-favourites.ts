export const SprykerFavouritesResponseMock = {
  data: [
    {
      type: 'wishlist-items',
      id: 'list-id',
      attributes: {
        name: 'name',
        numberOfItems: 2,
        createdAt: 'string',
        updatedAt: 'string'
      },
      links: {
        self: 'string'
      },
      relationships: {
        'wishlist-items': {
          data: [
            {
              type: 'string',
              id: 'string'
            }
          ]
        }
      }
    }
  ],
  links: { self: 'string' },
  included: [
    {
      type: 'string',
      id: 'some-id-1',
      attributes: { sku: 'string' },
      links: { self: 'string' }
    },
    {
      type: 'string',
      id: 'some-id-2',
      attributes: { sku: 'string' },
      links: { self: 'string' }
    }
  ]
}

export const GetFavouritesResponseMock = {
  id: 'list-id',
  items: ['some-id-1', 'some-id-2']
}
