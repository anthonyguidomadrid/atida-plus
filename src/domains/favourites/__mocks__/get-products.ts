export const productSkusMock = ['100000001', '100000002', '100000003']

export const productsMock = [
  {
    url: '/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas',
    name: 'Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas',
    enabled: true,
    brand: { code: 'atida', label: 'Atida' },
    sku: '100000001',
    format: { code: 'shampooCode', label: 'Shampoo' },
    image:
      'https://www.mifarma.es/media/catalog/product/cache/1/image/9d6758f7fb7476b09aa51a3fa7ce618a/1/8/180671_.jpg',
    productDatImage:
      'https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0',
    thumbnailImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    mediumImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    largeImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    productTileImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    productTileRetinaImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    description:
      '<p><strong>Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas 2+1</strong>\n</p><p>¡Recupera la fuerza y vitalidad de tu cabello con Forcapil!\n</p><p>Indicado para hombres y mujeres, este nutricosmético previene y combate la caída del pelo.\n</p><p>Fórmula exclusivamente con ingredientes 100% de origen natural.\n</p><p>Tiene una alta concentración en biotina, y su composición con aminoácidos, vitaminas, minerales y plantas favorece el crecimiento de la melena.\n</p><p>Se recomienda tomar en momentos en los que sufrimos una mayor pérdida de cabello, causada por estrés, menopausia, tras el parto, cambios de estación...\n</p><p><strong>Tratamiento para 3 meses: 2 + 1 de REGALO.</strong>\n</p>',
    unitVolume: {
      amount: 100,
      unit: 'MILLILITER',
      unitLabel: 'ml'
    },
    contentSizeFactor: 1,
    price: {
      value: 9999,
      currency: 'EUR'
    },
    pricePerUnit: {
      value: 9999,
      currency: 'EUR',
      unit: '100 ml'
    },
    rrp: {
      value: 9999,
      currency: 'EUR'
    },
    availability: 'AVAILABLE'
  },
  {
    url: '/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas',
    name: 'Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas',
    enabled: true,
    brand: { code: 'atida', label: 'Atida' },
    sku: '100000002',
    format: { code: 'shampooCode', label: 'Shampoo' },
    image:
      'https://www.mifarma.es/media/catalog/product/cache/1/image/9d6758f7fb7476b09aa51a3fa7ce618a/1/8/180671_.jpg',
    productDatImage:
      'https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0',
    thumbnailImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    mediumImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    largeImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    productTileImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    productTileRetinaImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    description:
      '<p><strong>Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas 2+1</strong>\n</p><p>¡Recupera la fuerza y vitalidad de tu cabello con Forcapil!\n</p><p>Indicado para hombres y mujeres, este nutricosmético previene y combate la caída del pelo.\n</p><p>Fórmula exclusivamente con ingredientes 100% de origen natural.\n</p><p>Tiene una alta concentración en biotina, y su composición con aminoácidos, vitaminas, minerales y plantas favorece el crecimiento de la melena.\n</p><p>Se recomienda tomar en momentos en los que sufrimos una mayor pérdida de cabello, causada por estrés, menopausia, tras el parto, cambios de estación...\n</p><p><strong>Tratamiento para 3 meses: 2 + 1 de REGALO.</strong>\n</p>',
    unitVolume: {
      amount: 100,
      unit: 'MILLILITER',
      unitLabel: 'ml'
    },
    contentSizeFactor: 1,
    price: {
      value: 9999,
      currency: 'EUR'
    },
    pricePerUnit: {
      value: 9999,
      currency: 'EUR',
      unit: '100 ml'
    },
    rrp: {
      value: 9999,
      currency: 'EUR'
    },
    availability: 'AVAILABLE'
  },
  {
    url: '/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas',
    name: 'Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas',
    enabled: true,
    brand: { code: 'atida', label: 'Atida' },
    sku: '100000003',
    format: { code: 'shampooCode', label: 'Shampoo' },
    image:
      'https://www.mifarma.es/media/catalog/product/cache/1/image/9d6758f7fb7476b09aa51a3fa7ce618a/1/8/180671_.jpg',
    productDatImage:
      'https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0',
    thumbnailImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    mediumImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    largeImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    productTileImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    productTileRetinaImage:
      'https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png',
    description:
      '<p><strong>Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas 2+1</strong>\n</p><p>¡Recupera la fuerza y vitalidad de tu cabello con Forcapil!\n</p><p>Indicado para hombres y mujeres, este nutricosmético previene y combate la caída del pelo.\n</p><p>Fórmula exclusivamente con ingredientes 100% de origen natural.\n</p><p>Tiene una alta concentración en biotina, y su composición con aminoácidos, vitaminas, minerales y plantas favorece el crecimiento de la melena.\n</p><p>Se recomienda tomar en momentos en los que sufrimos una mayor pérdida de cabello, causada por estrés, menopausia, tras el parto, cambios de estación...\n</p><p><strong>Tratamiento para 3 meses: 2 + 1 de REGALO.</strong>\n</p>',
    unitVolume: {
      amount: 100,
      unit: 'MILLILITER',
      unitLabel: 'ml'
    },
    contentSizeFactor: 1,
    price: {
      value: 9999,
      currency: 'EUR'
    },
    pricePerUnit: {
      value: 9999,
      currency: 'EUR',
      unit: '100 ml'
    },
    rrp: {
      value: 9999,
      currency: 'EUR'
    },
    availability: 'AVAILABLE'
  }
]
