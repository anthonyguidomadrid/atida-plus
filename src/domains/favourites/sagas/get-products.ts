import { AxiosResponse } from 'axios'
import type { SagaIterator } from 'redux-saga'
import { all, call, getContext, put, takeLatest } from 'redux-saga/effects'
import { createClient, logger } from '~helpers'
import {
  getFavouritesProductsTrigger,
  getFavouritesProductsSuccess,
  getFavouritesFulfill,
  getFavouritesProductsFailure
} from '../slices'
import type { PayloadAction } from '@reduxjs/toolkit'
import {
  FavouritesItemsIdsPayload,
  FavouritesWithProducts
} from '~domains/favourites/types'
import { getErrorMessage } from '~helpers/error'

function* doGetProducts({ payload }: PayloadAction<FavouritesItemsIdsPayload>) {
  try {
    const locale: string = yield getContext('locale')
    const client = createClient({ locale, addAnonymousCustomerUniqueId: false })
    const response: AxiosResponse<FavouritesWithProducts> = yield call(
      client.post,
      '/api/favourites/get-products',
      payload
    )

    yield put(getFavouritesProductsSuccess(response.data))
  } catch (error) {
    const errorMessages = getErrorMessage(error)
    logger.error(errorMessages)
    if (Array.isArray(errorMessages)) {
      yield put(
        getFavouritesProductsFailure({
          message: errorMessages[0]
        })
      )
    } else {
      yield put(
        getFavouritesProductsFailure({
          message: errorMessages
        })
      )
    }
  } finally {
    yield put(getFavouritesFulfill())
  }
}

export function* getProductsSaga(): SagaIterator {
  yield all([takeLatest(getFavouritesProductsTrigger, doGetProducts)])
}
