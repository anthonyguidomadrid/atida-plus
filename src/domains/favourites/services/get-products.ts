import { FeatureFlag } from '~config/constants/feature-flags'
import {
  createElasticSearchClient,
  getElasticSearchProductIndex
} from '~domains/elasticsearch/helpers/client'
import {
  ChannelPrice,
  ElasticSearchProduct,
  normalizeElasticSearchProduct
} from '~domains/product'
import { loadFeatureFlag } from '~helpers/server-only/featureFlagClient'
import { FavouritesWithProducts, FavouritesItemsIdsPayload } from '../types'

if (typeof window !== 'undefined')
  throw new Error('This file should not be called client-side')

export const getProducts = async (
  locale: string,
  data?: FavouritesItemsIdsPayload
): Promise<FavouritesWithProducts> => {
  if (!data?.skus?.length) {
    return data as FavouritesWithProducts
  }
  // TODO: Remove when the price channels on products are ready to be used
  const mockedProductPriceChannels = (await loadFeatureFlag(
    locale,
    FeatureFlag.MUTLICHANNEL_PRICES_MOCKED_PRODUCT_PRICE_CHANNEL
  )) as { channel_prices: ChannelPrice[] }

  const client = createElasticSearchClient()
  const response = await client.mget<{ docs: ElasticSearchProduct[] }>({
    index: getElasticSearchProductIndex(locale),
    body: {
      ids: data.skus.map(item => item)
    }
  })

  const products = response?.body?.docs?.map(doc =>
    normalizeElasticSearchProduct(
      locale,
      doc,
      data.sessionChannel,
      mockedProductPriceChannels?.channel_prices
    )
  )

  return products?.length ? products : []
}
