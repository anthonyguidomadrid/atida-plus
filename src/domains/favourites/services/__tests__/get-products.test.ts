/**
 * @jest-environment node
 */
// @ts-ignore
import { mget } from '@elastic/elasticsearch'
import { productsMock, productSkusMock } from '../../__mocks__/get-products'
import { getProducts } from '../get-products'

describe(getProducts, () => {
  beforeEach(() => {
    mget.mockResolvedValue(productsMock)
  })

  it('passes the correct index and skus to elasticsearch request', async () => {
    await getProducts('en-gb', {
      skus: productSkusMock,
      sessionChannel: { channel: 'amazon', sku: 'some-sku' }
    })
    expect(mget).toHaveBeenCalledWith({
      body: {
        ids: productSkusMock
      },
      index: 'index-en-gb'
    })
  })

  it('returns the normalized basket response with products', async () => {
    const response = await getProducts('en-gb', {
      skus: productSkusMock,
      sessionChannel: { channel: 'amazon', sku: 'some-sku' }
    })
    expect(response).toMatchSnapshot()
  })

  it('returns empty products', async () => {
    const response = await getProducts('en-gb', {
      skus: [],
      sessionChannel: undefined
    })
    expect(response).toMatchSnapshot()
  })
})
