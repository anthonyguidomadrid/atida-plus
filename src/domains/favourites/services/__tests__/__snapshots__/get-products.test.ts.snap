// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProducts returns empty products 1`] = `
Object {
  "sessionChannel": undefined,
  "skus": Array [],
}
`;

exports[`getProducts returns the normalized basket response with products 1`] = `Array []`;
